src/skin.cpp

```diff
@@ -496,6 +496,45 @@ void SkinInfillAreaComputation::generateRoofing(SliceLayerPart& part)
     }
 }
 
+void SkinInfillAreaComputation::generateInfillSupport(SliceMeshStorage& mesh)
+{
+    const coord_t layer_height = mesh.getSettingInMicrons("layer_height");
+    const double support_angle = mesh.getSettingInAngleRadians("infill_support_angle");
+    const double tan_angle = tan(support_angle) - 0.01;  //The X/Y component of the support angle. 0.01 to make 90 degrees work too.
+    const coord_t max_dist_from_lower_layer = tan_angle * layer_height; //Maximum horizontal distance that can be bridged.
+
+    for (int layer_idx = mesh.layers.size() - 2; layer_idx >= 0; layer_idx--)
+    {
+        SliceLayer& layer = mesh.layers[layer_idx];
+        SliceLayer& layer_above = mesh.layers[layer_idx + 1];
+        
+        Polygons inside_above;
+        Polygons infill_above;
+        for (SliceLayerPart& part_above : layer_above.parts)
+        {
+            inside_above.add(part_above.infill_area);
+            infill_above.add(part_above.getOwnInfillArea());
+        }
+
+        for (SliceLayerPart& part : layer.parts)
+        {
+            const Polygons& infill_area = part.infill_area;
+            if (infill_area.empty())
+            {
+                continue;
+            }
+
+            const Polygons unsupported = infill_area.offset(-max_dist_from_lower_layer);
+            const Polygons basic_overhang = unsupported.difference(inside_above);
+            const Polygons overhang_extented = basic_overhang.offset(max_dist_from_lower_layer + 50); // +100 for easier joining with support from layer above
```

Comment:
+100? :stuck_out_tongue:

Perhaps use `support_join_distance` instead? You'd be having a support setting that influences infill, but it probably needs to be equal in just about all cases anyway.

Reply:
I think it's counterintuitive to use the same option. It would then also have to be enabled even when support is not generated.

We could make it an extra option, but the whole idea of this PR was to have the functionality without introducing the millions of settings that support has.

If you find some cases which fail badly because of this magic value I could reconsider the implementation, but I'd rather just have it merged and tri s out by people.

Reply:
I'll just change the documentation then.